src/Microsoft.AspNetCore.Hosting.Abstractions/IHostingApplicationData.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore.Hosting
+{
+    public interface IHostingApplicationData
```

Comment:
I'm not sure I like this name. We have ContentRoot and WebRoot. We should stick to that paradigm.

Reply:
I had application data root but I hated it. @glennc had some more high level feedback, so we are pausing this for preview 1 until you're back and we can all discuss it, but it seems that you're ok with the approach I took? See https://github.com/aspnet/Hosting/issues/1294 for details on an alternative proposal